File: Python/A_Project/B_Snake_and_Food/MovementStrategyModule/ConcreteMovementStrategies/human_movement_strategy.py
from MovementStrategyModule.movement_strategy import MovementStrategy
from Utility.pair import Pair


class HumanMovementStrategy(MovementStrategy):
    def get_next_position(self, current_head, direction: str):
        row = current_head.row
        col = current_head.col
        if direction == "U":
            return Pair(row - 1, col)
        if direction == "D":
            return Pair(row + 1, col)
        if direction == "L":
            return Pair(row, col - 1)
        if direction == "R":
            return Pair(row, col + 1)
        return current_head

File: Ruby/A_Project/B_Snake_And_Food/MovementStrategyModule/ConcreteMovementStrategies/human_movement_strategy.rb
# frozen_string_literal: true

require_relative "../movement_strategy"
require_relative "../../Utility/pair"

class HumanMovementStrategy < MovementStrategy
  def get_next_position(current_head, direction)
    row = current_head.row
    col = current_head.col

    case direction
    when "U"
      Pair.new(row - 1, col)
    when "D"
      Pair.new(row + 1, col)
    when "L"
      Pair.new(row, col - 1)
    when "R"
      Pair.new(row, col + 1)
    else
      current_head
    end
  end
end
